src/ert/storage/local_ensemble.py

```diff
@@ -318,42 +347,62 @@ def _load_single_dataset(
                 f"No dataset '{group}' in storage for realization {realization}"
             ) from e
 
-    def _load_dataset(
+    def load_parameters(
         self,
         group: str,
-        realizations: Union[int, npt.NDArray[np.int_], None],
+        realizations: Union[int, Tuple[int], npt.NDArray[np.int_], None] = None,
     ) -> xr.Dataset:
-        if isinstance(realizations, int):
-            return self._load_single_dataset(group, realizations).isel(
-                realizations=0, drop=True
-            )
+        try:
+            self.open_unified_dataset(group)
+        except FileNotFoundError:
+            self._unify_parameters()
 
-        if realizations is None:
-            datasets = [
-                xr.open_dataset(p, engine="scipy")
-                for p in sorted(self.mount_point.glob(f"realization-*/{group}.nc"))
-            ]
-        else:
-            datasets = [self._load_single_dataset(group, i) for i in realizations]
-        return xr.combine_nested(datasets, "realizations")
+        try:
+            ds = self.open_unified_dataset(group)
+
+            if realizations is not None:
+                return ds.sel(
+                    realizations=(
+                        np.array(realizations)
+                        if type(realizations) is not int
+                        else realizations
+                    )
+                )
+            return ds
+        except (KeyError, FileNotFoundError) as e:
+            raise KeyError(
+                f"No dataset '{group}' in storage for realization {realizations}"
+            ) from e
 
-    def load_parameters(
-        self, group: str, realizations: Union[int, npt.NDArray[np.int_], None] = None
-    ) -> xr.Dataset:
-        return self._load_dataset(group, realizations)
+    def open_unified_dataset(self, key: str) -> xr.Dataset:
+        nc_path = self._path / f"{key}.nc"
+
+        if os.path.exists(nc_path):
+            return xr.open_dataset(nc_path)
+
+        raise FileNotFoundError(
+            f"Dataset file for group {key} not found (tried {key}.nc)"
+        )
 
     @lru_cache  # noqa: B019
     def load_responses(self, key: str, realizations: Tuple[int]) -> xr.Dataset:
-        if key not in self.experiment.response_configuration:
+        if key not in self.experiment.response_info:
             raise ValueError(f"{key} is not a response")
-        loaded = []
-        for realization in realizations:
-            input_path = self._realization_dir(realization) / f"{key}.nc"
-            if not input_path.exists():
-                raise KeyError(f"No response for key {key}, realization: {realization}")
-            ds = xr.open_dataset(input_path, engine="scipy")
-            loaded.append(ds)
-        return xr.combine_nested(loaded, concat_dim="realization")
+        try:
+            self.open_unified_dataset(key)
```

Comment:
Refactored down to open only once. The first open within the try-catch is to see if it is already unified. If it is not combined, it will make the combined dataset then. This "lazy" unify thing makes it possible to do it without having to explicitly tell `storage` from the outside when to unify (which assumes that it is used "sensibly" from the outside, not causing loads of re-computations of the unified dataset). Refactored this into one function to make its purpose more clear.